metis/lib/query_builder.rb

```diff
@@ -4,44 +4,51 @@
 class Metis
   class QueryBuilder
 
-    def initialize(base_query, params)
+    def initialize(base_query, params, limit = nil, offset = 0)
       @base_query = base_query
       @params = params
+      @limit = limit
+      @offset = offset
     end
 
     def build
+      result = @base_query
       @params.each do |param|
         next unless param.key?(:attribute) && param.key?(:predicate) && param.key?(:value)
 
         # Modify query behavior based on "type" of attribute
         case param[:attribute]
         when 'created_at', 'updated_at'
-          @base_query = @base_query.where{|o|
+          result = result.where{|o|
```

Comment:
Don't mutate the base_query variable -- use a local variable instance to ensure that successive calls to build don't continuously mutate all previous query builds.